src/main/jbake/content/blog/2016/09/19/hawkular-apm-on-msa.adoc

```diff
@@ -0,0 +1,163 @@
+be used as a drop-in replacement for ZipKin.
+
+Red Hat offers a link:http://developers.redhat.com/products/cdk/overview/[Container Developer Toolkit], or CDK, for its customers,
+allowing for a quick start on developing applications following the best practices on the microservices architecture.
+
+Along with the CDK, Red Hat publishes also a link:https://github.com/redhat-helloworld-msa["MSA - Hello World Microservices Architecture"],
+which demonstrates how applications might benefit from this architecture. It can also be used as a seed project for your own
+projects, as it shows how activities can be accomplished by using several technology stacks, from `nodejs/express` to `Wildfly Swarm`.
+
+For Hawkular APM, this offers a great way to test our APIs on different stacks, specially our ZipKin compatibility layer. Ideally,
+one would simply point the ZipKin URL setting to a Hawkular APM server and everything would just work.
+
+And that's what we are going to do on this post!
+
+== Getting started with CDK and MSA
+
+=== CDK and MSA
+To setup CDK and MSA, we suggest to follow the link:http://bit.ly/msainstructions[official instructions]. At the end of the process,
+you should have all the services running. Open the "frontend" service, and click around a bit. For instance:
+
+* Click on the "Browser as client" menu item, which will make the browser make Ajax calls directly to the API endpoints
+* Click on the "API Gateway" menu item: on this case, the browser makes a single call to the API Gateway, which in turn makes single API calls
+to each of the services
+* Click on "Service chaining" menu item: on this case, each service calls the "next", adding the results to its own payload.
+* It could be interesting to make a note about the hostnames, to later make sure that you have "new" Docker containers running.
+The default output for each of the services include the hostname. Concretely, the output `ola-7-9wa31` means that the service `ola` is
+running on the container with hostname `9wa31`.
+
+Open the ZipKin interface, and make sure that data is arriving there. Check also the different visualizations for the different operations.
+
+=== Hawkular APM
+We'll use the link:http://www.hawkular.org/blog/2016/07/14/hawkular-apm-openshift.html[previous post on running Hawkular APM on OpenShift]
+as base to install our Hawkular APM alongside with the other MSA examples. For this blog post, we'll prepare our own image based on the official
+Dockerfile for Hawkular APM. For that, fork and clone link:https://github.com/jboss-dockerfiles/hawkular[our repository] and run:
+
+```bash
+cd hawkular-apm-server
+oc new-build --binary --name=hawkular-apm-server -l app=hawkular-apm-server
+oc start-build hawkular-apm-server --from-dir=. --follow
+```
+
+Alternatively, the image from Docker Hub can be used, although networking issues are somewhat common:
```

Comment:
It has to download the image from Dockerhub, instead of building the image locally and uploading to the Vagrant image. At least on my case, it worked only on the third attempt, even though I have a very good internet connection.